File: 31-anagram/anagram.rb
class Anagram
  attr_reader :anagram

  def initialize(anagram)
    @anagram = anagram
  end

  def match(words)
    answer = []
    words.each do |word|
      if word.split("").sort == anagram.split("").sort
        answer << word
      end
    end
    answer
  end
end

File: exercism/javascript/anagram/anagram.js
Anagram = function(word){
  this.match = function(matches){
    answer = [];

    var normalize = function(input){
      return input.split("").sort().toString()
    }

    var compareArrays = function(first, second){
      return normalize(first) === normalize(second);
    };

    for(var i = 0; i < matches.length; i++){
      if (compareArrays(matches[i], word)){
        answer.push(matches[i]);
      }
    }
    return answer;
  }
};
